### DISCUSSION TITLE
Referencing dart value mapped by converter in drift query criteria

### DISCUSSION THREAD

**Discussion by adg-mh** [simolus3/drift#3159] [General]
Not sure exactly how this would work, but is it possible to somehow reference a dart value (mapped by converter for that column) in a drift query criteria? e.g.

```
SELECT a, b 
FROM SomeTable
WHERE b = `WhateverEnum.pending`
```

Currently I just use the integer value and add a comment, but that requires care in refactoring.
(I know that dart enums have special handling elsewhere, but I'll note for my use cases it's not always a dart enum)

**Comment by simolus3** [comment]
We have no support to embed fixed Dart expressions in SQL queries at the moment (an interesting idea though). I will mention though that you can embed arbitrary Dart expressions evaluated at runtime by using [Dart components](https://drift.simonbinder.eu/docs/sql-api/drift_files/#dart-components-in-sql). That allows you to define the statement as e.g.

```SQL
SELECT a, b 
FROM SomeTable
WHERE $condition
```

And then call it with `thatQuery(condition: (row) => row.b.equalsValue(WhateverEnum.pending))`.
